detection_rules/config.py

```diff
@@ -150,16 +151,38 @@ class RulesConfig:
     exception_dir: Optional[Path] = None
 
 
+def check_path(content: dict, keys: tuple, content_name: str):
+    """Check if a dict key path exists in the supplied content."""
+    try:
+        reduce(lambda d, key: d[key], keys, content)
+    except KeyError:
+        raise ValueError(f"Missing {'.'.join(keys)} in {content_name} contents.")
+
+
+def validate_package_contents(package_contents: dict):
+    """Validate the contents of the package for required fields."""
+
+    paths = [
+        ('name',),
+        ('registry_data', 'conditions'),
+        ('registry_data', 'conditions', 'kibana.version')
+    ]
+
+    for path in paths:
+        check_path(package_contents, path, "packages.yml")
+
+
 @cached
 def parse_rules_config(path: Optional[Path] = None) -> RulesConfig:
     """Parse the _config.yaml file for default or custom rules."""
     if path:
         assert path.exists(), f'rules config file does not exist: {path}'
         loaded = yaml.safe_load(path.read_text())
-    elif CUSTOM_RULES_DIR:
-        path = Path(CUSTOM_RULES_DIR) / '_config.yaml'
-        assert path.exists(), f'_config.yaml file missing in {CUSTOM_RULES_DIR}'
+    elif CUSTOM_RULES_DIR and (path := Path(CUSTOM_RULES_DIR) / '_config.yaml').exists():
         loaded = yaml.safe_load(path.read_text())
+        if 'configuration_details' in loaded or not loaded:
+            raise SystemExit(loaded['configuration_details'])
```

Comment:
```suggestion
            raise ValueError(f'Missing required content for config file: {path} (default contents detected')
```

This stays consistent with raising a `ValueError` for missing contents (should be the same).

Then in `__main__.py`

```python
try:
    RULES_CONFIG = parse_rules_config()
except SpecificErrorRaised as e:
    print('There was an error loading the _config.yml file', file=sys.stderr)  # or click.echo
    sys.exit(not 0) # or raise
```

Reply:
Based on the discussion in the original [thread](https://github.com/elastic/detection-rules/pull/3653#discussion_r1594157686). We will have to make this change in multiple changes, which IMO is not the best/cleanest approach. So to meet in the middle, I'm raising a SystemExit vs (sys.exit or the try except approach you mentioned).

Reply:
no you dont - main is the first load and the whole point of putting it there. It forces an early load and an early failure - the rest are _cached_ loads

Reply:
also rather than checking for `configuration_details` here, it would be more resilient to check for an absence of the required keys